testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/TestClassProvider.java

```diff
@@ -0,0 +1,82 @@
+package org.keycloak.testsuite.arquillian;
+
+import com.sun.net.httpserver.HttpContext;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpServer;
+import org.apache.commons.io.IOUtils;
+import org.jboss.logging.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+
+public class TestClassProvider {
+
+    private static final Logger LOGGER = Logger.getLogger(TestClassProvider.class);
+
+    public static final String[] PERMITTED_PACKAGES = new String[] {
+            "/org/keycloak/testsuite",
+            "/org/junit",
+            "/org/hamcrest",
+            "/org/keycloak/admin/client",
+            "/org/jboss/resteasy/client",
+            "/org/jboss/arquillian",
+            "/org/jboss/shrinkwrap",
+            "/org/openqa/selenium"
+    };
+
+    private static TestClassProvider instance;
+    private final HttpServer server;
+
+    public static TestClassProvider getInstance() throws IOException {
+        if (instance == null) {
+            instance = new TestClassProvider();
+        }
+        return instance;
+    }
+
+    private TestClassProvider() throws IOException {
+        InetSocketAddress address = new InetSocketAddress(8500);
+        server = HttpServer.create(address, 0);
+        HttpContext context = server.createContext("/");
+        context.setHandler(TestClassProvider::handleRequest);
+        server.start();
+
+        LOGGER.infov("Started test class provider {0}", address.toString());
+    }
+
+    public void stop() {
+        server.stop(0);
+    }
+
+    private static void handleRequest(HttpExchange exchange) throws IOException {
+        String className = exchange.getRequestURI().toString();
+
+        LOGGER.infov("Request: {0}", className);
+
+        InputStream is = TestClassProvider.class.getResourceAsStream(className);
```

Comment:
```suggestion
        try (InputStream is = TestClassProvider.class.getResourceAsStream(className)) {
            ...
        }
```

Reply:
This is what happens when a prototype becomes something more ;)

Completely changed now with switch to Undertow